### PR TITLE
RI-7901: support deleting folder in tree view

### PR DESCRIPTION
**Pull request by ArtemHoruzhenko**
# RI-7901: Support Deleting Folder in Tree View

## Summary

Adds a delete button to folder nodes in the tree view that opens the Bulk Actions panel pre-populated with the folder's pattern.

References #5382 

### Features
- Delete icon on folder nodes (appears on hover)
- Opens Bulk Actions panel with pattern `folder:*` pre-filled
- Respects current key type filter
- Shows folder-specific key count in summary

### Disabled States
- Multiple delimiters configured → tooltip explains limitation
- Non-printable characters in folder name → prevents unreliable bulk operations

### State Management
- New bulk delete state: `filter`, `search`, `keyCount`
- Isolated from main browser state to prevent race conditions
- Local key removal from Redux after successful delete (tree recalculates without API refresh)

### UX Improvements
- "Download report" checkbox unchecked by default
- Division by zero protection when `scanned === 0`
- Previous bulk delete state reset when clicking new folder

## Testing

- Unit tests for all new reducers and components
- Folder delete button behavior tests
- Edge case coverage (unprintable chars, multiple delimiters)

---

> [!NOTE]
> **Medium Risk**
> Touches bulk delete flow/state and socket event handling; mistakes could lead to incorrect UI state, stale key lists, or triggering deletes with unintended patterns/filters.
> 
> **Overview**
> Adds a **folder-level delete action in Tree View**: folder nodes now show a hover delete icon that opens the Bulk Actions panel prefilled with the folder pattern (e.g. `folder:*`), preserves the current key-type filter, and emits new telemetry (`TREE_VIEW_FOLDER_DELETE_CLICKED`). The delete affordance is *disabled* (with explanatory tooltips) when multiple delimiters are configured or folder names contain unprintable characters.
> 
> Refactors bulk delete to use dedicated Redux state (`bulkDelete.search`, `bulkDelete.filter`, `bulkDelete.keyCount`) that is kept in sync from search/filter UI, and updates bulk delete UI to use `keyCount` for folder deletes plus safer “expected amount” calculations (including `scanned === 0`). Bulk delete socket handling now registers the overview listener before `Create` and, on completion, prunes locally cached keys via new `deleteKeysByPattern` to avoid requiring an immediate refresh.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c0f7fd7502b4fcd38ebeecd73722b057d4d46487. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: redisinsight/ui/src/pages/browser/components/virtual-tree/components/Node/Node.styles.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  React.HTMLAttributes<HTMLDivElement>
+>`
+  display: inline-block;
+  width: 86px;
+  min-width: 86px;
+  text-align: right;
+  transition: transform ease 0.3s;
+`
+
+export const FolderKeyCount = styled.div<React.HTMLAttributes<HTMLDivElement>>`
```
Comment (valkirilov) [inline redisinsight/ui/src/pages/browser/components/virtual-tree/components/Node/Node.styles.ts:14]: Lol, that type 😀

**Review by valkirilov — APPROVED** [review]
